prov/usnic/src/usdf_fabric.c

```diff
@@ -189,6 +190,42 @@ usdf_fill_addr_info(struct fi_info *fi, uint32_t addr_format,
 	return ret;		// fi_freeinfo() in caller frees all
 }
 
+void build_devname(struct usd_device_attrs *dap, char **bufp)
+{
+	struct in_addr in;
+	char *addrnetw;
+	char devname[40];
+
+	//USDF_DBG("NOW: ipaddr_be:(%d) prefixlen:(%d)\n",
+	//	  dap->uda_ipaddr_be, dap->uda_prefixlen);
+
+	//in.s_addr = dap->uda_ipaddr_be;
+	//addrnetw = inet_ntoa(in);
+	//USDF_DBG("NOW: ipaddr_be(converted):(%s)\n",
+	//	 addrnetw);
+
+	//uint32_t nmask = (0xFFFFFFFF >> (32 - dap->uda_prefixlen)) & 0xFFFFFFFF; // switch >> to << for LE
+	//USDF_DBG("NOW: netmask(computed): %#010x\n",
+	//	 nmask);
+
+	//in.s_addr = dap->uda_ipaddr_be & nmask;
+	//addrnetw = inet_ntoa(in);
+	//USDF_DBG("NOW: ipaddr_be(network):(%s)\n",
+	//	 addrnetw);
+
+	//USDF_DBG("NOW: netmask_be (embed): %#010x\n",
+	//	 dap->uda_netmask_be);
+
+	in.s_addr = dap->uda_ipaddr_be & dap->uda_netmask_be;
+	addrnetw = inet_ntoa(in);
+	//USDF_DBG("NOW: ipaddr_be(network2):(%s)\n",
+	//	 addrnetw);
+
+	snprintf(devname, 40, "%s/%d", addrnetw, dap->uda_prefixlen);
```

Comment:
Meh.  I think a hard-coded 40 for the buffer length in the declaration/definition is fine -- you know that 40 bytes is definitely longer than you will need for this simple string, the max form of which is `xxx.xxx.xxx.xxx/xxx`.